substrate-node/pallets/pallet-tfgrid/src/grid_migration.rs

```diff
@@ -243,14 +245,28 @@ pub fn migrate_farms<T: Config>() -> frame_support::weights::Weight {
             }
         }
 
-        let replaced_farm_name = farm::replace_farm_name_spaces_with_underscores(&farm.name);
-        let name = match <T as Config>::FarmName::try_from(replaced_farm_name) {
-            Ok(n) => n,
-            Err(_) => {
-                info!("invalid farm name, skipping updating farm {:?} ...", k);
-                return None;
-            }
-        };
+        let mut farm_name = farm.name;
+        let truncated = farm_name.len() > <T as Config>::MaxFarmNameLength::get() as usize;
+        if truncated {
+            info!("farm name length exceeded, truncating farm name");
+            farm_name.truncate(<T as Config>::MaxFarmNameLength::get() as usize);
+        }
+
+        let mut replaced_farm_name = farm::replace_farm_name_invalid_characters(&farm_name);
+        let name = <T as Config>::FarmName::try_from(replaced_farm_name.clone()).unwrap_or({
```

Comment:
Use unwrap_or_else here. unwrap_or is eager, meaning it will evaluate the value in every loop iteration, regardless of whether it is used or not.